### ISSUE TITLE
Assert.Throws shows double stack trace

### ISSUE THREAD

**Issue by bradwilson** [xunit/xunit#2353]
Repro from #2351:

```csharp
using System;
using Xunit;

public class MyTestClass
{
    private void DoSomething() => throw new Exception("This is not an InvalidOperationException");

    [Fact]
    public void AssertThrows()
    {
        Assert.Throws<InvalidOperationException>(() => DoSomething());
    }

    [Fact]
    public void RecordException()
    {
        var ex = Record.Exception(() => DoSomething());

        Assert.IsType<InvalidOperationException>(ex);
    }
}
```

Results:

```
xUnit.net v3 In-Process Runner v0.1.0-pre.383+850587bdbc (64-bit .NET 5.0.8)
  Discovering: xunit-2351
  Discovered:  xunit-2351
  Starting:    xunit-2351
    MyTestClass.AssertThrows [FAIL]
      Assert.Throws() Failure
      Expected: typeof(System.InvalidOperationException)
      Actual:   typeof(System.Exception): This is not an InvalidOperationException
      ---- System.Exception : This is not an InvalidOperationException
      Stack Trace:
        UnitTest1.cs(6,0): at MyTestClass.DoSomething()
        UnitTest1.cs(11,0): at MyTestClass.<Test1>b__1_0()
        ----- Inner Stack Trace -----
        UnitTest1.cs(6,0): at MyTestClass.DoSomething()
        UnitTest1.cs(11,0): at MyTestClass.<Test1>b__1_0()
    MyTestClass.RecordException [FAIL]
      Assert.IsType() Failure
      Expected: System.InvalidOperationException
      Actual:   System.Exception
      Stack Trace:
        UnitTest1.cs(19,0): at MyTestClass.Test2()
  Finished:    xunit-2351
=== TEST EXECUTION SUMMARY ===
   xunit-2351  Total: 2, Errors: 0, Failed: 2, Skipped: 0, Time: 0.067s
```

Observations:

* `Assert.Throws` shows `typeof(TypeName)` whereas `Assert.IsType` just shows `TypeName`
* `Assert.Throws` shows doubled information: The type, the message, and the stack trace all appear to be doubled up.

**Comment by Xcodo** [comment]
This should have been closed by #2356

**Comment by bradwilson** [comment]
Yep, you're right. Thanks!
